src/global/ValueId.h

```diff
@@ -69,7 +71,30 @@ class ValueId {
   /// This exception is thrown if we try to store a value of an index type
   /// (VocabIndex, LocalVocabIndex, TextRecordIndex) that is larger than
   /// `maxIndex`.
-  struct IndexTooLargeException : public std::exception {};
+  struct IndexTooLargeException : public std::exception {
+    private:
+      
+      ad_utility::source_location _location;
+
+      T _tooBigValue;
+
+      std::string _errorMessage;
+
+    public:
+
+      IndexTooLargeException(T tooBigValue,
+          ad_utility::source_location s = ad_utility::source_location::current()):
+          _location(s), _tooBigValue(tooBigValue) {
+        std::stringstream exceptionMessage;
+        exceptionMessage << _location.file_name() << ", line " << _location.line()
+          << ": The given value " << _tooBigValue << " is bigger than what the maxIndex of ValueId allows.";
+        _errorMessage = exceptionMessage.str();
```

Comment:
Slight problem with that, `absl::StrCat` needs files to be linked with `absl::strings` and this helper file is only a header. So we would need to find all the files, that use it and link them with the `absol::strings` library.

Reply:
Apparently, there will now be a new PR, which changes the CMake, so that there will be a default library, which will be linked with all compiled files in the project. Once that is done, this issue can be done.